conda_build/source.py

```diff
@@ -117,16 +119,15 @@ def download_to_cache(cache_folder, recipe_path, source_dict, verbose=False):
             raise RuntimeError("Could not download %s" % url)
 
     hashed = None
-    for tp in ("md5", "sha1", "sha256"):
-        if tp in source_dict:
-            expected_hash = source_dict[tp]
-            hashed = hashsum_file(path, tp)
-            if expected_hash != hashed:
-                rm_rf(path)
-                raise RuntimeError(
-                    f"{tp.upper()} mismatch: '{hashed}' != '{expected_hash}'"
-                )
-            break
+
+    for tp in set(source_dict).intersection(ACCEPTED_HASH_TYPES):
+        expected_hash = source_dict[tp]
+        hashed = hashsum_file(path, tp)
+        if expected_hash != hashed:
+            rm_rf(path)
+            raise RuntimeError(
+                f"{tp.upper()} mismatch: '{hashed}' != '{expected_hash}'"
+            )
```

Comment:
Renaming two character variable for a more descriptive one:

```suggestion
    for hash_type in set(source_dict).intersection(ACCEPTED_HASH_TYPES):
        expected_hash = source_dict[hash_type]
        hashed = hashsum_file(path, hash_type)
        if expected_hash != hashed:
            rm_rf(path)
            raise RuntimeError(
                f"{hash_type.upper()} mismatch: '{hashed}' != '{expected_hash}'"
            )
```